lake/commits/actions.go

```diff
@@ -39,31 +41,22 @@ func (a *Add) String() string {
 	return fmt.Sprintf("ADD %s", a.Segment)
 }
 
-type CommitMessage struct {
-	Commit  ksuid.KSUID `zng:"commit"`
+type Commit struct {
+	ID      ksuid.KSUID `zng:"id"`
+	Parent  ksuid.KSUID `zng:"parent"`
+	Retries uint8       `zng:"retries"`
```

Comment:
What the story on this?

Reply:
I will move this comment from branch.go to actions.go...
```
		// Note that we store the number of retries in the final commit
		// object.  This will allow easily introspection of optimistic
		// locking problems under high commit load by simply issuing
		// a meta-query and looking at the retry count in the persisted
		// commit objects.  If/when this is a problem, we could add
		// pessimistic locking mechanisms alongside the optimistic approach.
```

Reply:
Great idea!

Reply:
(I hadn't read that comment. :grin:)